lib/pages/discover_page.dart

```diff
@@ -61,18 +69,36 @@ class DiscoverPage extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    DiscoverBloc bloc = DiscoverBloc(
+      eventService: Provider.of<EventService>(
+        context,
+        listen: false,
```

Comment:
Added listen: false so that changes to the service to not cause rebuilds.  Read docs for more info on that.